### PR TITLE
ci: enable golangci-lint

### PR DESCRIPTION
**Pull request by ccoVeille**
Currently 43 errors, more rules could be enabled.

But let's start with these ones.

Fixes:
- #221

### CODE REVIEW

**Comment by ccoVeille** [comment]
if you want to test locally simply launch

```console
golangci-lint run ./...
```

Some linters are able to apply fix via golangci-lint, if you want to apply fix use this
```console
golangci-lint run -fix ./...
```

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -51,6 +51,12 @@ jobs:
         with:
           go-version-file: 'go.mod'
 
+      - name: Generate embed file
+        run: go generate ./...
+
+      - name: Launch golangci-lint
+        uses: golangci/golangci-lint-action@v6
+
```
Comment (nobe4) [inline .github/workflows/ci.yml:59]: Can you move this into a new job instead?
This way, `golang-ci-lint` failing doesn't prevent the `lint` job from running.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -51,6 +51,12 @@ jobs:
         with:
           go-version-file: 'go.mod'
 
+      - name: Generate embed file
+        run: go generate ./...
+
+      - name: Launch golangci-lint
```
Comment (nobe4) [inline .github/workflows/ci.yml:57]:
```suggestion
      - name: golangci-lint
```

File: .golangci.yml [hunk 3]
```diff
@@ -0,0 +1,162 @@
+---
```
Comment (nobe4) [inline .golangci.yml:1]: I would remove all comments from this file.

Leaving just at the top of the file:
```yaml
# Ref: https://golangci-lint.run/usage/configuration/
# Inspired by https://github.com/ccoVeille/golangci-lint-config-examples/
```

For documentation and attribution.

**Review by nobe4 — APPROVED** [review]
Looks appropriate! Thanks!

I'll merge this with the :x: and will fix in a subsequent PR.

**Pull request merged**